### ISSUE TITLE
Create a stacked column % chart with a vertical legend to the right

### ISSUE THREAD

**Issue by tfavorite** [infor-design/enterprise#4587]
**Is your feature request related to a problem or use case? Please describe.**
There are several chart donuts that might do better as a single stacked column of 100%, with the legend to the right. Legends frequently sort in the same order as the chart items. Users don't often get to see that because legends are placed beneath, and wrap around, making following the order less intuitive for users (and eating up vertical space).

Here, we envision a legend that is a simple, vertical column of colors and labels that is more easily parsed by the user (and saves vertical space). See attached wire (below) for layout.

**Describe the solution you'd like**
An option to put legend on the side.

**Describe alternatives you've considered**
The only option right now seems to be putting the legend on the bottom.

**Additional context**
<img width="517" alt="TO-status-stacked-column-right-legend" src="https://user-images.githubusercontent.com/347235/98560109-6969ee00-2275-11eb-9829-28a2b1f7aff7.png">
And what application do you work on? Infor Nexus

**Comment by tmcconechy** [comment]
I really like this suggestion. Sometimes the pie chart breaks down with a lot of elements or smaller elements.
We have this and its somewhat similar https://master-enterprise.demo.design.infor.com/components/bullet/example-index.html in a way

**Comment by tfavorite** [comment]
That one is good to know about too! Will consider that one in other cases, though I think right now our designer is looking for a stacked chart due to its compactness

**Comment by alex-infor** [comment]
Hello! I'm the designer requesting this, for use in two projects I'm working on. The crucial bits are having the column + legend to the right for improved usability, as well as reducing the vertical footprint on the page. Do we know how close it is to being available?

**Comment by tmcconechy** [comment]
cc @inforandy do we have a designer that's able to approve or tweak this?
@alex-infor at this point i've just brought in our designers to make sure this is ok to build. So its not that close.

But i do see the advantage of this, especially some cases where it could replace a pie chart with awkwardly small slices so im personally of for it. 

Adding to next sprint review for further discussion

**Comment by alex-infor** [comment]
Thanks for the update :-) I'm presenting to the design team on color and data vis usability this afternoon, so perhaps that will help.

**Comment by kevinwhitedesign** [comment]
@tmcconechy @inforandy @alex-infor Hey guys! Yeah I totally see the value of the chart placement as described in this scenario. Generally I think it makes sense to have placement from LEFT > RIGHT and TOP > BOTTOM in order of viewing patterns (localization flipped in some cases). 

We have some examples of both placements described (right & bottom) on our design documentation for the pie chart https://design.infor.com/product/data-visualization/pie-chart. I think it ultimately depends on the scenario / given layout.

As far as the Stacked column percentage chart I agree I think could be valuable as you say for:
- Potentially more efficient readability (@alex-infor would love to see any references/data/research around this)
- Flexible to vertically constrained layouts

**Comment by kevinwhitedesign** [comment]
Clarifying: Just saying that UI content is usually ordered in viewing priority from left to right OR top to bottom. I think it's correct in this case to put the most important stuff first (chart) and then supporting stuff second (legend).

**Comment by inforandy** [comment]
Thanks @kevinwhitedesign, and just clarifying for myself in case, we are implying that since reading patterns in western languages tend towards TOP,LEFT -> BOTTOM, RIGHT, then a secondary piece of information like a chart legend should be able to display in both the BOTTOM and the RIGHT without issue, correct?

**Comment by inforandy** [comment]
nvm, answered it for me (YES), thanks!

**Comment by tmcconechy** [comment]
The legend is somewhat secondary so i think its correct and normal its either to the right (considering reading order) or below. In this case it fits on the right so this design works i think.

**Comment by alex-infor** [comment]
I'm not sure what research you're looking for, because most of this is based in core design principles and well-known aspects of how the brain parses data. The question here is primarily one of alignment, proximity, and order of content (category alignment, if you will), as well as chart type. 

- You have a single, 100% column of stacked content with an axis to provide context, which data suggests should work better than a pie chart. The only study that shows a pie chart beating a bar chart is when the bar chart's axis is removed; the only studies that show stacked bars are poor is when you have multiple stacked bar charts compared to each other, which is not the case here. 
- Shared alignment makes it easier to compare things (this is a basic design tenet). If you have a legend that wraps around beneath a vertical stack, the wrapped legend is going to be just that much harder to align with the single column than a vertical legend immediately to the right. 
- Order alignment is also a consideration. It's not just placing the content closer, it's placing it in the order of the chart content, further improving the ability of the brain to "get" what the user is viewing. 
- Placing the legend to the right also places it closer to this particular chart, so it makes the action of parsing what's going on easier.
- We want to conserve space vertically and horizontally. Since this is a single stack, it can easily accommodate a legend beside it without adding horizontal space; wrapping a legend beneath it left to right, left to right, left to right takes up more vertical space.

I don't get asked to break down design insights at this level often. Hope this helps make things clearer.

**Comment by kevinwhitedesign** [comment]
Hey @alex-infor well said here "well-known aspects of how the brain parses data" yess. 
Was just wondering if you had any docs you were referencing around these valid points you're making. Could benefit IDS in terms of: adding a resources section to our documentation pages, internal explorations on the subjects, and as reference as we think about our next steps in data visualization. Even though these may be design basic tenets, I'm always seeking to lean on existing materials/research to help others share an understanding they consume our components & documentation.

I'm with you on most of this. I think it could be valuable introducing a concept of 'shared alignment' for chart segments and legend labels _if it actually improves readability_ - although @tmcconechy may have to speak on feasibility there. Maybe if there's a technical challenge there you guys can go ahead and explore it. 

Attached a little exploration around this, let me know if this is tracking. 
<img width="1262" alt="Screen Shot 2020-12-15 at 4 27 23 PM" src="https://user-images.githubusercontent.com/6478704/102274826-7d4ae480-3ef2-11eb-834e-f0694b4163bf.png">

**Comment by tmcconechy** [comment]
For sure - one though is a lot of the time the labels are long or wrap. What would happen in those cases? For that reason i think the first one makes it a bit easier so its in order. If the lines are small or a lot of them there it will be hard to keep the labels lined up with the data

**Comment by kevinwhitedesign** [comment]
Yeah I have some design hesitance on the right 2 examples. I wonder if it actually will increase readability with everything so spread out and sporadic in positioning?

**Comment by alex-infor** [comment]
Re: research. When I taught this, I always recommended Stephen Few's blog Perceptual Edge to students. It's a great springboard, and Few will often have a post connecting the dots on different nuances. 

Re: Legend explorations. I was not suggesting reinventing the legend itself. I don't know if that's wise - I see what you're trying to do, actually aligning each segment to its legend item, but I think that risks:
- The awkward versioning on the far right
- Confusion between whether this is a legend or data

Re: order feasibility. When I used D3 and dimple D3, the legends almost always presented in the same order as the chart data. So hopefully it's as simple as trying it. I am not seeking to add a ton of dev effort to this. 

Re: actually improving readability. I think any designer can reasonably connect the dots on things like shared alignment, consistent information design (the order question), etc. (This is why we use things like tables, instead of listing separate paragraphs of data.) What reasons do you see that make this unusable, that you ask?

**Comment by tmcconechy** [comment]
ok then i think the "Always Consistent Spacing" example should be good - and not too much dev effort compared to lining up the data and worrying about wrapping and different sized bar sections

**Comment by kevinwhitedesign** [comment]
Ha, I misread your comment around 'Shared Alignment'. I thought you were advocating for the 2nd/3rd example in my explorations. 

And order feasibility shouldn't be an issue though I don't want to speak for Tim. I was concerned about the css required to align the segments & legend item.

**Comment by alex-infor** [comment]
Ah, got it! Yeah, that would be weird, and I'd question it myself :-)

**Comment by alex-infor** [comment]
Re: building a good data vis reference library. The Few site is good, and I've also seen some good work in Nightingale, a Medium publication that popped up last year. https://medium.com/nightingale/welcome-to-nightingale-1089f0e73a00

**Comment by kevinwhitedesign** [comment]
Awesome I just found Stephen Few's piece on 100% stacked bars too actually - https://www.perceptualedge.com/blog/?p=2239

**Comment by alex-infor** [comment]
That's a great post! When they work, when they are awful and horrible, and alternative strategies where appropriate.

**Comment by tmcconechy** [comment]
This is sort of a good example too on why this is a good idea vs a pie chart http://www.perceptualedge.com/example5.php - i've seen many cases like this of developers asking to "fix" the pie chart. Pretty much same concept except ours is flipped here.

**Comment by tmcconechy** [comment]
Note to self
- try to put one legend on right / left on current chart
- make sure the legend position is a feature in new chart
- build this design in both old and new

**Comment by kevinwhitedesign** [comment]
This is done?

**Comment by tmcconechy** [comment]
No we pushed this out to a future sprint but will do it soon.

**Comment by alex-infor** [comment]
Thanks, this is good to hear! We have dashboards-in-progress that will be the better for this option.
